hail/python/hail/expr/blockmatrix_type.py

```diff
@@ -0,0 +1,72 @@
+from hail.typecheck import typecheck_method
+
+
+class tblockmatrix(object):
+    @staticmethod
+    def _from_java(jtbm):
+        return tblockmatrix(jtbm.nRows(), jtbm.nCols(), jtbm.blockSize())
+
+    @typecheck_method(n_rows=int, n_cols=int, block_size=int)
+    def __init__(self, n_rows, n_cols, block_size):
+        self.n_rows = n_rows
+        self.n_cols = n_cols
+        self.block_size = block_size
+
+    def __eq__(self, other):
+        return isinstance(other, tblockmatrix) and \
+               self.n_rows == other.n_rows and \
+               self.n_cols == other.n_cols and \
+               self.block_size == other.block_size
+
+    def __hash__(self):
+        return 43 + hash(str(self))
+
+    def __repr__(self):
+        return f'tblockmatrix(n_rows={self.n_rows!r}, n_cols={self.n_cols!r}, block_size={self.block_size!r})'
+
+    def __str__(self):
+        return f'blockmatrix {{n_rows={self.n_rows}, n_cols={self.n_cols}, block_size={self.block_size}}}'
```

Comment:
We use : instead of = in our pretty-printed types.

Reply:
Done